### PR TITLE
Ramp WebGL integration

### LINKED ISSUE #637: Ramp Fiat integration - Web
- [ ] Connect Unity side with Ramp
- [ ] Create a new class that inherits from RampIntegrationBase.
- [ ] In RampIntegraionSample add appropriate preprocessor directive to create the instance.
- [ ] Make sure that it's possible to subscribe from Unity to the native API event for when off ramp sale happened, when  onramppurchase happened.

### PR DESCRIPTION
**Pull request by oleksandrchainsafe**
**This PR Introduces:**

1. A Ramp wrapper for WebGL builds. 
2. New architecture for the whole Ramp package.
3. iOS implementation updates to stay in line with the new architectural decisions.
4. Updated samples.

**Known issues:**

The sample scene only works in WebGL builds currently. We'll have to figure out a new approach for logging in users for our packages' samples before we can fix this issue.

Closes #637

### CODE REVIEW

**Comment by oleksandrchainsafe** [comment]
Merge after #634

File: Packages/io.chainsafe.web3-unity.exchangers/Runtime/Plugins/Ramp.jslib [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                    stringToNewUTF8(sale.crypto.assetInfo.symbol),
+                    stringToNewUTF8(sale.crypto.assetInfo.type),
+                    sale.fiat.amount,
+                    stringToNewUTF8(sale.fiat.currencySymbol),
+                    );
+            })
+            .show();
+    },
+    
+    // todo use this if stringToNewUTF8() doesn't work
```
Comment (kantagara) [inline Packages/io.chainsafe.web3-unity.exchangers/Runtime/Plugins/Ramp.jslib:96]: We can probably delete this if you have tested it and it works.

File: src/UnitySampleProject/Assets/RampExchangerSample.cs
Comment (kantagara) [inline src/UnitySampleProject/Assets/RampExchangerSample.cs:1]: What's the difference between this and RampSamples

**Review by kantagara — APPROVED** [review]
Looks good, I'm approving this but will have to put some stuff in the corresponding platform branches (iOS and Android)

File: Packages/io.chainsafe.web3-unity.exchangers/Runtime/Plugins/Ramp.jslib (hunk 1 above)
Comment (oleksandrchainsafe) [reply Packages/io.chainsafe.web3-unity.exchangers/Runtime/Plugins/Ramp.jslib:114]: Haven't tested it yet. Can't fix the preceding code.

**Comment by empea-careercriminal** [comment]
@creeppak may I ask you to own this one by breaking down the refactoring this will invoke in a separate issue, branch, PR, and make #648 dependent on it? 🙏🏼

**Comment by oleksandrchainsafe** [comment]
> @creeppak may I ask you to own this one by breaking down the refactoring this will invoke in a separate issue, branch, PR, and make #648 dependent on it? 🙏🏼

Yes. Done✅

**Review by sneakzttv — APPROVED** [review]

**Review by juans-chainsafe — CHANGES_REQUESTED** [review]
Is working in WebGL and iOS with Web3Auth login, I tested on ramp with test data and tried to test off ramp with test data but some weird error with (I think) my location happened:
```
Error response: {
    "payoutMethod": [
        "Payout method not available in user legal Regime"
    ]
} 
```
But I dont think we can do something about it.

What is failing on our side:

- WebGL build with Wallet connect, after I login, I get the following error when I try to click "Try On-Ramp" or "Try Off-Ramp":
```
webgl2.framework.js.br:3 ArgumentNullException: Value cannot be null.
Parameter name: format
  at System.String.FormatHelper (System.IFormatProvider provider, System.String format, System.ParamsArray args) [0x00000] in <00000000000000000000000000000000>:0 
--- End of stack trace from previous location where exception was thrown ---
```

- Looks like we removed SampleMain from the Scenes to Build in Build Settings, it was on purpose?
Screenshot 2024-01-24 at 13 35 58

**Review by juans-chainsafe — APPROVED** [review]
Working in iOS build as well!

**Pull request merged**